### ISSUE TITLE
Support pyparsing v3 (AttributeError downcaseTokens)

### ISSUE THREAD

**Issue by mborsetti** [httplib2/httplib2#207]
There appears to be a use of a [discontinued feature](https://pyparsing-docs.readthedocs.io/en/latest/whats_new_in_3_0_0.html?highlight=downcaseTokens#other-discontinued-features) from the `pyparsing` library:

```
Traceback (most recent call last):
[...]
  File "/usr/local/lib/python3.9/dist-packages/httplib2/__init__.py", line 52, in <module>
    from . import auth
  File "/usr/local/lib/python3.9/dist-packages/httplib2/auth.py", line 20, in <module>
    auth_param_name = token.copy().setName("auth-param-name").addParseAction(pp.downcaseTokens)
AttributeError: module 'pyparsing' has no attribute 'downcaseTokens'
```

**Comment by mborsetti** [comment]
> > For the long run, could replacing `pp.downcaseTokens` with `pp.pyparsing_common.downcase_tokens` be a possible solution? That fixed it in my fresh virtualenv, it seems a bit too easy though, so I might be missing something.
> 
> It fixes import, but parser doesn't work.
> 
> Please try following:
> 
> ```shell
> git clone https://github.com/httplib2/httplib2 tmp207
> cd tmp207
> git apply <<END
> diff --git a/python3/httplib2/auth.py b/python3/httplib2/auth.py
> index 84b5831..9fe69f2 100644
> --- a/python3/httplib2/auth.py
> +++ b/python3/httplib2/auth.py
> @@ -5,6 +5,11 @@ import pyparsing as pp
> 
>  from .error import *
> 
> +try:
> +    downcaseTokens = pp.common.downcaseTokens
> +except AttributeError:
> +    downcaseTokens = pp.downcaseTokens
> +
>  UNQUOTE_PAIRS = re.compile(r"\\(.)")
>  unquote = lambda s, l, t: UNQUOTE_PAIRS.sub(r"\1", t[0][1:-1])
> 
> @@ -17,7 +22,7 @@ token68 = pp.Combine(pp.Word("-._~+/" + pp.nums + pp.alphas) + pp.Optional(pp.Wo
>  )
> 
>  quoted_string = pp.dblQuotedString.copy().setName("quoted-string").setParseAction(unquote)
> -auth_param_name = token.copy().setName("auth-param-name").addParseAction(pp.downcaseTokens)
> +auth_param_name = token.copy().setName("auth-param-name").addParseAction(downcaseTokens)
>  auth_param = auth_param_name + pp.Suppress("=") + (quoted_string | token)
>  params = pp.Dict(pp.delimitedList(pp.Group(auth_param)))
> END
> python3 -m venv venv
> venv/bin/pip install -e . -r requirements-test.txt
> venv/bin/pip install 'pyparsing>=3'
> venv/bin/pytest -sv tests/test_auth.py
> ```

@temoto Thanks; this seems to work.

**Comment by temoto** [comment]
It definitely doesn't work. Subscribe to recent pyparsing issue.

**Comment by temoto** [comment]
Fixed in `httplib2>=0.20.2`
